core/transfer/local/transfer.go

```diff
@@ -54,8 +56,14 @@ func NewTransferService(cs content.Store, is images.Store, tc TransferConfig) tr
 	if tc.MaxConcurrentUploadedLayers > 0 {
 		ts.limiterU = semaphore.NewWeighted(int64(tc.MaxConcurrentUploadedLayers))
 	}
+	if tc.MaxConcurrentDownloadOperations > 0 {
+		ts.limiterOperationD = semaphore.NewWeighted(int64(tc.MaxConcurrentDownloadOperations))
+	}
 	if tc.MaxConcurrentDownloads > 0 {
 		ts.limiterD = semaphore.NewWeighted(int64(tc.MaxConcurrentDownloads))
+	} else {
+		// default to 3 dl at a time
+		ts.limiterD = semaphore.NewWeighted(int64(3))
```

Comment:
why do we need a default 3 semaphore now? can we still keep the previous behavior if tc.MaxConcurrentDownloads == 0?

Reply:
The default was 3, so, using this as the default, ex:

```
--max-concurrent-downloads int            Set the max concurrent downloads for each pull (default 3)
```

I changed this behaviour because, before this PR `max-concurrent-downloads` was creating a semaphore that was limiting the number of operations (copy/unpack/else) AND the number of downloads. With this PR, we are still limiting to 3 downloads at a time, but the operation semaphore is another one.

Reply:
That operations semaphore is not limited by default `MaxConcurrentDownloadOperations`, and maybe I should put it to 1 to keep the old behaviour ? I think that 0 is also fine here, for a setting with 3 downloads at a time.

Reply:
After a bunch of re-reading, I can put back the default. But the user will have to not set `MaxConcurrentDownloadsPerLayer`, otherwise we might use way too much ram